firefox-ios/Client/Coordinators/Browser/BrowserCoordinator.swift

```diff
@@ -620,9 +617,25 @@ class BrowserCoordinator: BaseCoordinator,
 
     // MARK: - WindowEventCoordinator
 
-    func coordinatorWindowWillClose() {
-        // This cleanup is necessary to ensure BVC and other components are properly released.
-        browserViewController.contentContainer.subviews.forEach { $0.removeFromSuperview() }
-        browserViewController.removeFromParent()
+    func coordinatorHandleWindowEvent(event: WindowEvent, uuid: WindowUUID) {
+        switch event {
+        case .windowWillClose:
+            // Was the closed window the window for this browser?
+            guard uuid == windowUUID else { return }
+            // If so, perform add'tl cleanup required so BVC and other components are properly released.
+            browserViewController.contentContainer.subviews.forEach { $0.removeFromSuperview() }
+            browserViewController.removeFromParent()
+        case .libraryOpened:
+            // If the library was opened for this browser's window, we can ignore
+            // Otherwise, auto-close the panel in this window [FXIOS-8095]
+            guard uuid != windowUUID else { return }
+
+            guard let libraryCoordinator = childCoordinators[LibraryCoordinator.self] else { return }
+            let browserPresentedVC = router.navigationController.presentedViewController
+            let rootVC = (browserPresentedVC as? DismissableNavigationViewController)?.viewControllers.first
+            if rootVC === libraryCoordinator.router.rootViewController {
+                router.dismiss(animated: true, completion: nil)
+            }
```

Comment:
Note: this was the most straightforward solution I could see for detecting if the library panel is presented (and closing it), but if there is a better / more convenient way of doing this I'm happy to update this code or revisit it. (cc @OrlaM @lmarceau)